specification/cognitiveservices/data-plane/Personalizer/preview/v1.0/Personalizer.json

```diff
@@ -0,0 +1,1042 @@
+          }
+        },
+        "excludedActions": {
+          "description": "The set of action ids to exclude from ranking.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "eventId": {
+          "description": "Optionally pass an eventId that uniquely identifies this Rank event. \r\nIf null, the service generates a unique eventId. The eventId will be used for\r\nassociating this request with its reward, as well as seeding the pseudo-random\r\ngenerator when making a Personalizer call.",
+          "type": "string"
+        },
+        "deferActivation": {
+          "description": "Send false if the user will see the rank results, therefore\r\nPersonalizer will expect a Reward call, otherwise it will assign the default\r\nReward to the event. Send true if it is possible the user will not see the\r\nrank results, because the page is rendering later, or the Rank results may be\r\noverridden by code further downstream.",
+          "default": false,
+          "type": "boolean"
+        }
+      }
+    },
+    "RankableAction": {
+      "description": "An action with it\u0027s associated features used for ranking.",
+      "required": [
+        "id",
+        "features"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Id of the action.",
+          "type": "string"
+        },
+        "features": {
+          "description": "List of dictionaries containing features.",
+          "type": "array",
+          "items": {
+            "type": "object"
+          }
+        }
+      }
+    },
+    "RankResponse": {
+      "description": "A resulting ordered list of actions that result from a rank request.",
+      "type": "object",
+      "properties": {
+        "ranking": {
+          "description": "The calculated ranking for the current request.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RankedAction"
+          }
+        },
+        "eventId": {
+          "description": "The eventId for the round trip from request to response.",
+          "type": "string"
+        },
+        "rewardActionId": {
+          "description": "The action chosen by the Personalizer service. This is the action for which to report the reward. This might not be the \r\nfirst found in \u0027ranking\u0027 if an action in the request in first position was part of the excluded ids.",
+          "type": "string"
+        }
+      }
+    },
+    "RankedAction": {
+      "description": "A ranked action with its resulting probability.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Id of the action",
+          "type": "string"
+        },
+        "probability": {
+          "format": "float",
+          "description": "Probability of the action",
+          "type": "number"
+        }
+      }
+    },
+    "ContainerStatus": {
+      "type": "object",
```

Comment:
Please check which of the model properties are marked as `readOnly`, applies to all model definitions.

Reply:
marked  some response properties as readonly